src/libraries/System.IO.MemoryMappedFiles/src/System/IO/MemoryMappedFiles/MemoryMappedFile.cs

```diff
@@ -154,12 +154,6 @@ public static MemoryMappedFile CreateFromFile(string path, FileMode mode, string
                 throw new ArgumentException(SR.Argument_EmptyFile);
             }
 
-            if (access == MemoryMappedFileAccess.Read && capacity > fileStream.Length)
```

Comment:
Lines 143-146 in this file:
```cs
if (access == MemoryMappedFileAccess.Write)
{
    throw new ArgumentException(SR.Argument_NewMMFWriteAccessNotAllowed, nameof(access));
}
```

Should you remove those lines, since you deferred that exception to `CreateCore`?

Reply:
I tried to explain it in the comment above @carlossanlop. Let me know if that makes. If not, we can chat next week.

Reply:
Thanks for the explanation in person, @wfurt. As discussed, the Write-only exception should be reverted to its original state to avoid affecting existing behavior.